activemodel/lib/active_model/validations/numericality.rb

```diff
@@ -93,7 +93,7 @@ def parse_as_number(raw_value)
         elsif is_integer?(raw_value)
           raw_value.to_i
         elsif is_decimal?(raw_value) && !is_hexadecimal_literal?(raw_value)
-          BigDecimal(raw_value)
+          BigDecimal(raw_value.to_s)
```

Comment:
Before `d126c0d` `to_s` was never called, so why do we need to call now? And, what in `d126c0d` was making this work since we didn't have `to_s`?

Reply:
I guess my commit message could have been more clear 😅 

BigDecimal will raise a TypeError if the initial is neither Integer, Float, Rational, BigDecimal nor String. The use of BigDecimal was introduced in `d126c0d`.


The change made in https://github.com/rails/rails/pull/35010/files#diff-58f66170fe2bce77e8a8b3c69b77a0d6R109
isn't related to the same commit, but to https://github.com/rails/rails/pull/34693

Reply:
I tried to reproduce the `Kernel.Float` behavior and it doesn't implicitly call `to_s`.

```
>> class Foo
>>   def to_s
>>     "1"
>>     end
>>   end
=> :to_s
>> Kernel.Float(Foo.new)
Traceback (most recent call last):
        5: from /Users/rafaelfranca/.rbenv/versions/2.6.0/bin/irb:23:in `<main>'
        4: from /Users/rafaelfranca/.rbenv/versions/2.6.0/bin/irb:23:in `load'
        3: from /Users/rafaelfranca/.rbenv/versions/2.6.0/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):6
        1: from (irb):6:in `Float'
TypeError (can't convert Foo into Float)
```

Reply:
Thanks for double checking you are right I got confused. From the Ruby doc

> Kernel.Float(arg) → float
Returns arg converted to a float. Numeric types are converted directly, the rest are converted using arg.to_f.

To keep compatibility, that would mean we'd have to call `BigDecimal(raw_value.to_f, Float::DIG + 1` (we can't convert a Float without precision)

That being said, to check if the raw_value is a decimal we are calling `to_s` on it already https://github.com/rails/rails/blob/3f0c7c5580e67120b2a42552c3624b4e552aa721/activemodel/lib/active_model/validations/numericality.rb#L105 
Don't we want to keep consistent and also call `to_s` when converting to a BigDecimal ?

Reply:
I was hopping by understanding how this was working in the past we could decide what is the best way to fix this. I'm not against calling to_s, but it seems like to_s was not what was used in the past.